doc/ImplementingApplications/Steps2IntegrateElasticsearch/Steps2IntegrateElasticsearch.md

```diff
@@ -0,0 +1,255 @@
+## Integrate Elasticsearch (ES)
+
+Disclaimer: This guide is not meant to be a manual for ES itself,
+it only covers areas implemented in ApplicationPattern and guides how to use
+the class ElasticsearchService in your application. For more information about
+Elasticsearch, go to [official documentation](https://www.elastic.co/).
+
+### What is ES and why should my application use it?
+
+ES is a distributed, free and open search and analytics engine for all types of data, including textual, numerical, geospatial, structured, and unstructured. Your application should consider using it, if:
+- it will store large amounts of data
+- it will store data often
+
+### Brief introduction to ES buzzwords
+
+Basic building block in ES is `document`. By this word, ES understands one item of `data`. 
+Documents have own `_id` and `_source`, where the actual data is stored. Documents are grouped
+into `indices`. One index groups data that are structured in the same way. Multiple indices
+can be grouped under an index `alias`. Index alias can be assigned a `service policy`. 
+Service policy is a set of rules, that determine when should be an index rolled over, deleted
+or do any other action specified in the policy. Index alias is tied to a service policy by an
+`index template`. This template gives it's properties to each index within the assigned index
+alias, including, but not limited to service policy, mapping and other settings. In order to execute requests towards ES, an ES
+Javascript `client` is used. See [documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/7.17/api-reference.html).
+
+### ElasticsearchService implementation in ApplicationPattern
+
+The version of Elasticsearch used in ApplicationPattern is 7.17 and so all the documentation from official website should be
+looked at with this version in mind.
+
+#### Configuration
+
+Application configuration should contain at least three LTPs.
+
+**ES client** itself with API key and index alias. If your application needs to use more aliases, you need to have more client LTPs. API key should be configured on ES directly.
+```
+{
```

Comment:
it would be better if we have a uuid in generalized pattern in ApplicationPattern . example : xx-1-0-0-es-c-es-1-0-0-000. 'xx' can be used for application-id in a general format. 'alt' seems more specific to ALT application. Or we can copy the es-c, http-c or tcp-c instance from config file of [AP](https://github.com/openBackhaul/ApplicationPattern/blob/develop/spec/ApplicationPattern%2Bconfig.json)